quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java

```diff
@@ -761,8 +761,16 @@ boolean exec() {
             task.doPark(ex == SuspendExecution.YIELD); // now we can complete parking
 
             assert ppa == null || ex == SuspendExecution.PARK; // can't have postParkActions on yield
-            if (ppa != null)
-                ppa.run(this);
+            if (ppa != null) {
+                // Redirect any exceptions thrown by the user code provided in the post-park action, as exceptions
+                // which propagate out of here may not always be caught and printed.
+                try {
```

Comment:
Hi. I don't understand this. This code is within a try/catch block that does the exact same handling. How can an exception thrown in ppa not be handled?